File: ruby/strategy/DuckSimulator/behavior/fly_behavior.rb
class FlyBehavior
  def fly
  end
end

class FlyWithWings < FlyBehavior

  def initialize(duck_name)
    @duck_name = duck_name
  end

  def fly
    super
    puts(@duck_name + " is flying with wings")
  end

end

class FlyNoWay < FlyBehavior

  def initialize(duck_name)
    @duck_name = duck_name
  end

  def fly
    super
    puts(@duck_name + " is not flying")
  end

end

File: javascript/strategy/DuckSimulator/behavior/FlyBehavior.js

export class FlyBehavior {
    fly() {}
}

export class FlyWithWings extends FlyBehavior {

    // private vars
    #duck_name

    constructor(duck_name) {
        super();
        this.#duck_name = duck_name
    }

    fly() {
        super.fly();
        console.log(this.#duck_name + " is flying with wings")
    }
}

export class FlyNoWay extends FlyBehavior {

    // private vars
    #duck_name

    constructor(duck_name) {
        super();
        this.#duck_name = duck_name
    }

    fly() {
        super.fly();
        console.log(this.#duck_name + " is not flying")
    }
}